### PR TITLE
Additional support for Mysql 

### PR DESCRIPTION
**Pull request by svigne1**
1. To use, Mysql, one has to change the config.json file, and change the following fields from 
   this
   "db_name" : "sqlite3",
   "db_path" : "gophish.db",
   to
   "db_name" : "mysql",
   "db_path" : "root:@(:3307)/gophish?charset=utf8&parseTime=True&loc=Local",
   where the format is 
   username:password@(host:port)/database?charset=utf8&parseTime=True&loc=Local
2. To create NEW migration files, (in the developer side), using "goose create" one can specify the path db/db_sqlite3 to "goose create" command using -path option.
   and run it again, giving db/db_mysql as the path now. ( before that, db/db_mysql/dbconf.yml needs to have proper value for field open)

### CODE REVIEW

**Comment by jordan-wright** [comment]
Thank you for this! I'll need to spin up the infrastructure to test this and make sure things are as clean as possible. In the mean time, I'll add some comments on some different areas.

Thank you again for taking this on - it's much appreciated.

File: db/db_mysql/migrations/20160118194630_init.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+CREATE TABLE IF NOT EXISTS results (id integer primary key auto_increment,campaign_id bigint,user_id bigint,r_id varchar(255),email varchar(255),first_name varchar(255),last_name varchar(255),status varchar(255) NOT NULL ,ip varchar(255),latitude real,longitude real );
+CREATE TABLE IF NOT EXISTS pages (id integer primary key auto_increment,user_id bigint,name varchar(255),html varchar(255),modified_date datetime );
+CREATE TABLE IF NOT EXISTS groups (id integer primary key auto_increment,user_id bigint,name varchar(255),modified_date datetime );
+CREATE TABLE IF NOT EXISTS group_targets (group_id bigint,target_id bigint );
+CREATE TABLE IF NOT EXISTS events (id integer primary key auto_increment,campaign_id bigint,email varchar(255),time datetime,message varchar(255) );
+CREATE TABLE IF NOT EXISTS campaigns (id integer primary key auto_increment,user_id bigint,name varchar(255) NOT NULL ,created_date datetime,completed_date datetime,template_id bigint,page_id bigint,status varchar(255),url varchar(255) );
+CREATE TABLE IF NOT EXISTS attachments (id integer primary key auto_increment,template_id bigint,content varchar(255),type varchar(255),name varchar(255) );
```
Comment (jordan-wright) [inline db/db_mysql/migrations/20160118194630_init.sql]: The "content" field should probably be a `BLOB` or `TEXT` - varchar is limiting in Mysql.

File: db/db_mysql/migrations/20160118194630_init.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+CREATE TABLE IF NOT EXISTS users (id integer primary key auto_increment,username varchar(255) NOT NULL UNIQUE,hash varchar(255),api_key varchar(255) NOT NULL UNIQUE );
+CREATE TABLE IF NOT EXISTS templates (id integer primary key auto_increment,user_id bigint,name varchar(255),subject varchar(255),text varchar(255),html varchar(255),modified_date datetime );
+CREATE TABLE IF NOT EXISTS targets (id integer primary key auto_increment,first_name varchar(255),last_name varchar(255),email varchar(255),position varchar(255) );
+CREATE TABLE IF NOT EXISTS smtp (smtp_id integer primary key auto_increment,campaign_id bigint,host varchar(255),username varchar(255),from_address varchar(255) );
+CREATE TABLE IF NOT EXISTS results (id integer primary key auto_increment,campaign_id bigint,user_id bigint,r_id varchar(255),email varchar(255),first_name varchar(255),last_name varchar(255),status varchar(255) NOT NULL ,ip varchar(255),latitude real,longitude real );
+CREATE TABLE IF NOT EXISTS pages (id integer primary key auto_increment,user_id bigint,name varchar(255),html varchar(255),modified_date datetime );
```
Comment (jordan-wright) [inline db/db_mysql/migrations/20160118194630_init.sql]: The "html" field should probably be a `BLOB` or `TEXT` - varchar is limiting in Mysql.

File: db/db_mysql/migrations/20160118194630_init.sql [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+CREATE TABLE IF NOT EXISTS users (id integer primary key auto_increment,username varchar(255) NOT NULL UNIQUE,hash varchar(255),api_key varchar(255) NOT NULL UNIQUE );
+CREATE TABLE IF NOT EXISTS templates (id integer primary key auto_increment,user_id bigint,name varchar(255),subject varchar(255),text varchar(255),html varchar(255),modified_date datetime );
```
Comment (jordan-wright) [inline db/db_mysql/migrations/20160118194630_init.sql]: The "html" and "text" fields should probably be a `BLOB` or `TEXT` - varchar is limiting in Mysql.

**Comment by svigne1** [comment]
Hi Jordan-wright,

Because of the line 17 in 20160227180335_0.1.2_store-smtp-settings.sql,
The code will not work in Mysql version less than 5.6.5

Please read this stackoverflow answer http://stackoverflow.com/a/168832/1499149
I will see what statement would suffice in the previous versions, and make the changes you mentioned in the above comments

**Comment by m4dw0lf** [comment]
Guys,

This feature is already working ?

**Comment by svigne1** [comment]
yeah, the code works fine for Mysql > 5.6.5

**Comment by Vanhecke** [comment]
I changed my config as stated in step 1 but the gophish application crashes when I launch it.
`runtime error: invalid memory address or nil pointer dereference`

I'm using the latest source from the master branch.

**Comment by svigne1** [comment]
This is still a pull request. It hasn't been merged yet. So, The master branch doesn't have this feature yet.

**Comment by jordan-wright** [comment]
I've gone ahead and merged this into a branch so I can do some testing, but everything is looking good so far! If all the testing looks good, then I'll merge into master.

Thanks again for your patience and for working on this!

**Pull request merged**